apolloschurchapp/src/index.js

```diff
@@ -77,72 +97,79 @@ const LandingToAuth = () => {
 const { Navigator, Screen } = createNativeStackNavigator();
 
 const App = () => (
-  <ThemeProvider themeInput={customTheme} iconInput={customIcons}>
-    <BackgroundView>
-      <AppStatusBar />
-      <ThemedNavigationContainer
-        containerRef={NavigationService.setTopLevelNavigator}
-        onReady={NavigationService.setIsReady}
-      >
-        <Providers>
-          <Navigator
-            screenOptions={{ headerShown: false, stackPresentation: 'modal' }}
+  <Themer theme={customTheme} icons={customIcons}>
+    <SafeAreaProvider>
+      <BottomSheetModalProvider>
```

Comment:
@conrad-vanl we won't lose them, I added them here. I think it's better because it's more explicit and you know what the various contexts are. Plus it avoids `recompose.nest`

Reply:
Yeah but this adds _more_ code to the template project, getting further away from template 0 right? Like now if we add a new provider, all of our frontend apps have to be updated. We can get rid of `nest()` inside of ui-kit if that's a concern.

Reply:
I personally think it's a bad abstraction for us to export another package's context provider. Likely _all_ the providers in the template will eventually be exported from NPM, but being explicit here is a better way to do it.

Reply:
for more context here our `LayoutProvider` is simply `SafeAreaProvider` re-exported

Reply:
> I personally think it's a bad abstraction for us to export another package's context provider.

Why?

currently, this is what ui-kit does, and to use ui-kit as that package is intended, we should import the `Provider` from it and render it here, not render out the providers separately. I don't think I'm comfortable approving splitting out to separate Providers unless we decide to change how UI-Kit works.

And still, I still argue that this is additional code and maintenance needed. If ui-kit adds another provider that is needed, you have to modify each app's code to incorporate that provider. This requires additional code ownership on each app.... I think that's the primary argument for the abstraction.

Reply:
Would you prefer I export a new `UIProviders` component from `ui-kit` without the old `ThemeProvider`?